### ISSUE TITLE
Leading slashes in routes with catch-all

### ISSUE THREAD

**Issue by mladedav** [ibraheemdev/matchit#42]
When I try to add routes to the router, it allows paths not starting with a forward slash, [except when a catch-all is used](https://github.com/ibraheemdev/matchit/blob/master/src/tree.rs#L277-L280).

This seems deliberate but I don't see why that is or why other routes not starting with a slash are allowed.

I might be missing something but I'd say that there should only be a slash right before the catch-all or it should be at the start of the route. So `*all` and `foo/*all` are both ok but `foo/bar*all` is not.

Here are some examples of what currently works and what doesn't.

```
let mut router = matchit::Router::new();
router.insert("a", ()).unwrap();
router.insert(":a", ()).unwrap();
router.insert("b/:b", ()).unwrap();
// Only the following two return errors
router.insert("c/*c", ()).unwrap();
router.insert("*d", ()).unwrap();
```

I'd be happy to provide a PR, but I want to check first if there are some considerations I'm missing.

**Comment by ibraheemdev** [comment]
The reason is that a catch-all parameters used to include the slash. Now that they don't this seems reasonable to allow.

**Comment by mladedav** [comment]
Hi @ibraheemdev, I got around to this but I've seen in some other issues you already have the matching reworked locally so I wonder if contributing here would not be helpful?

**Comment by ibraheemdev** [comment]
@mladedav I implemented this as part of [0.8](https://github.com/ibraheemdev/matchit/releases/tag/v0.8.0). Thanks for following up.
